### PR TITLE
Listener: record growth events for Identity and Marketplace

### PR DESCRIPTION
**Pull request by franckc**
First pull request? Read our [guide to contributing](http://docs.originprotocol.com/#contributing)

### Description:

Update the event listener to record entries in the growth_event table when there are Identity or Marketplace actions.

I did quite a bit of refactoring in the listener to make it more modular. The new structure is as follow:
 - listener.js: config, setup and main event loop
 - handler.js: receives an event, dispatches it to the proper handler
 - handler_identity.js: processes identity events
 - handler_marketplace.js: processes marketplace events
 - growth.js: implements method for inserting events in growth_event table

Other things to note:
 - replaced all console.log statements with logplease.
 - deprecated code for indexing user and offer in Elastic (we don't need that since we won't search that data). As a separate PR, I'll implement indexing user data in the DB.

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any displayed ETH addresses with [`formattedAddress`](https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/utils/user.js#L15-L17)
- [ ] Wrap any new text/strings for translation
- [ ] Run `npm run translations` if there are any changes to translated strings
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

### CODE REVIEW

**Comment by franckc** [comment]
@sparrowDom Still WIP (I mostly need to add unit tests), but would love to get some early feedback on this PR... Thanks !

File: origin-discovery/src/listener/handler_identity.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+   * @param {{blockNumber: number, logIndex: number}} blockInfo
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _recordGrowthAttestationEvents(user, blockInfo) {
+    const topicToEventType = {
+      'facebook': GrowthEventTypes.FacebookAttestationPublished,
+      'airbnb': GrowthEventTypes.AirbnbAttestationPublished,
+      'twitter': GrowthEventTypes.TwitterAttestationPublished,
+      'phone': GrowthEventTypes.PhoneAttestationPublished
```
Comment (sparrowDom) [inline origin-discovery/src/listener/handler_identity.js]: Should we also include the `email` attestation here?

Comment (franckc) [reply origin-discovery/src/listener/handler_identity.js]: Yes ! good catch

**Comment by DanielVF** [comment]
I can't comment on the growth engine side of it, but I love overall layout of this.

File: origin-discovery/src/listener/growth.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * @param {string} ethAddress
+ * @param {GrwothEventTypes} eventType
+ * @param {string} customId
+ * @param {Object} data
+ * @returns {Promise<void>}
+ */
+async function insertGrowthEvent(ethAddress, eventType, customId, data) {
+  // Check there isn't't already an event of same type.
+  const pastEvents = await db.GrowthEvent.findAll({
+    where: {
```
[the remark this answers is not in the document]
Comment (sparrowDom) [reply origin-discovery/src/listener/growth.js]: Nice thinking, the `custom id field` might also come in handy in the frontend in the future.

**Comment by sparrowDom** [comment]
Ahh sorry for double post. Seems you can not delete review message

File: origin-discovery/src/listener/handler_identity.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+   * @param {Object} user - Origin js user model object.
+   * @param {{blockNumber: number, logIndex: number}} blockInfo
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _recordGrowthAttestationEvents(user, blockInfo) {
+    user.attestations.forEach(async attestation => {
+      const eventType = AttestationTopicToEventType[attestation.topic]
+      if (!eventType) {
+        return
```
[the remark this answers is not in the document]
Comment (franckc) [reply origin-discovery/src/listener/handler_identity.js]: Yes, good point. Added.

**Comment by franckc** [comment]
Thanks for the prompt review ! Let me add the extra checks you mentioned in the unit test then I'll merge this into master...

**Pull request merged**